sw/host/opentitanlib/src/transport/qemu/mod.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{ensure, Context, Result};
+use std::cell::RefCell;
+use std::io::{self, ErrorKind};
+use std::net::{TcpStream, ToSocketAddrs};
+use std::rc::Rc;
+use std::thread;
+use std::time::{Duration, Instant};
+
+use crate::io::spi::Target;
+use crate::transport::{
+    Capabilities, Capability, Transport, TransportError, TransportInterfaceType,
+};
+
+pub mod spi;
+
+/// QEMU startup options.
+pub struct Options {
+    /// The QEMU host.
+    pub host: String,
+    /// The TCP port on the QEMU VM to connect to the SPI device CharDev.
+    pub spidev_port: u16,
+}
+
+pub struct Inner {
+    spi: Option<Rc<dyn Target>>,
+    stream: TcpStream,
+}
+
+impl Inner {
+    fn new(stream: TcpStream) -> Self {
+        Self {
+            spi: None,
+            stream: stream
+        }
+    }
+}
+
+pub struct QEMU {
```

Comment:
Nit:
```suggestion
pub struct Qemu {
```
Rust's style for acronyms is (maybe controversially) like this

Reply:
Yes, I noticed that too, the trouble is that is collides with how QEMU wants to be named:
```pl
# checkpatch.pl@2918
ERROR("use QEMU instead of Qemu or QEmu\n" . $herecurr);
```
_i.e._ QEMU would reject any patch with an incorrect spelling. It won't end up in QEMU upstream in anycase, but what takes precedence in this case? 
Should we spell it the Rust way or the QEMU way? 🤔 
There is a similar rule for [RISC-V](https://riscv.org/about/risc-v-branding-guidelines/) BTW, what is the Rust policy in this case?

Reply:
Not sure to understand the relationship between CI failure and my changes.

Reply:
Sorry to have missed this, if QEMU has a policy that they want it always spelled in capitals, then maybe we can accommodate for that (especially if it's some kind of trademark policy), otherwise I would lean towards using the Rust style within Rust code.

The CI failure looks unrelated, I don't think anything in the opentitan-private jobs would be affected by this change.

You may need to rebase to get it to rerun.